state/machine.go

```diff
@@ -2020,6 +2026,58 @@ func (m *Machine) UpdateMachineSeries(series string, force bool) error {
 	return errors.Annotatef(err, "cannot update series for %q to %s", m, series)
 }
 
+// CreateUpgradeSeriesPrepareLock create a prepare lock for series upgrade. If
+// this item exists in the database for a given machine it indicates that a
+// machine's operating system is being upgraded for one series to another (e.g. xenial to bionic).
+func (m *Machine) CreateUpgradeSeriesPrepareLock() error {
+	locked, err := m.IsLocked()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if locked {
+		return errors.Errorf("machine %q is already locked for upgrade", m)
+	}
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt > 0 {
+			if err := m.Refresh(); err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+
```

Comment:
should we check to see if the machine still exists? and life is set to Alive?  Adding corresponding asserts in the txn?

Reply:
yes